tests/models/test_openai.py

```diff
@@ -3325,3 +3325,53 @@ async def test_openai_reasoning_in_thinking_tags(allow_model_requests: None):
 """,
         }
     )
+
+
+def test_has_dict_typed_params_simple_dict():
+    """Test detection of simple dict[str, str] type."""
+    from pydantic_ai.models.openai import _has_dict_typed_params  # pyright: ignore[reportPrivateUsage]
+
+    schema = {'properties': {'my_dict': {'type': 'object', 'additionalProperties': {'type': 'string'}}}}
+    assert _has_dict_typed_params(schema) is True
+
+
+def test_has_dict_typed_params_nested_dict():
+    """Test detection of nested dict types."""
+    from pydantic_ai.models.openai import _has_dict_typed_params  # pyright: ignore[reportPrivateUsage]
+
+    schema = {
+        'properties': {
+            'nested': {'type': 'object', 'properties': {'inner_dict': {'type': 'object', 'additionalProperties': True}}}
+        }
+    }
+    assert _has_dict_typed_params(schema) is True
+
+
+def test_has_dict_typed_params_array_of_dicts():
+    """Test detection of list[dict[str, int]] type."""
+    from pydantic_ai.models.openai import _has_dict_typed_params  # pyright: ignore[reportPrivateUsage]
+
+    schema = {
+        'properties': {
+            'dict_list': {'type': 'array', 'items': {'type': 'object', 'additionalProperties': {'type': 'integer'}}}
+        }
+    }
```

Comment:
I'd prefer to test this with a tool function that actually has a `dict` argument, or a `BaseModel` argument that itself has a `dict` field, instead of testing the schemas directly.

So maybe can we build an agent with a tool like that, then run it, and test that a warning was emitted?